### DISCUSSION TITLE
How exactly will lightning configuration work if SupportLightning is set to false?

### DISCUSSION THREAD

**Discussion by hidaloop** [btcpayserver/btcpayserver#3868] [Technical Support] [answered]
Hello!
I have a question that I failed to resolve myself, thus I decided to come here in hope for clarification.
How exactly will lightning protocol configuration work for a particular store if `SupportLightning` variable in `BTCPayNetwork` instance is set to false?

Basically my question stems from the fact, that I tried to gently switch the lightning option off for my altcoin. I assumed that there could be a nice way to either do it programmatically or via Server Settings, instead of brute force and comment out html pieces XD

It seems that there are appear to be two triggers:
1. `BTCPayNetwork.SupportLightning` = false in `InitAltcoin` function
2. Server Settings > Policies > Allow non-admins to use the internal lightning node in their stores = **OFF**

This does remove Lightning from wallets section (Fig. 1)
<img width="211" alt="Fig. 1" src="https://user-images.githubusercontent.com/33467744/173575029-f1aaa2c9-658f-4942-baca-94d3243137bd.png"> (_Fig. 1_)

Yet it allows for Lightning integration from: 
- Setttings > Integrations > Lightning Address (Fig. 2)
- Freshly created store Dashboard (Fig. 3)
<img width="1038" alt="Fig. 2" src="https://user-images.githubusercontent.com/33467744/173575178-8b5f8437-0169-490b-92c2-1f509e727435.png"> (_Fig. 2_)
<img width="1792" alt="Fig. 3" src="https://user-images.githubusercontent.com/33467744/173575554-70c79c75-d181-4509-af94-d89f41f7ba94.png"> (_Fig. 3_)

Am I missing some understanding and it indeed should be there? Or should it rather be disabled as well, and this is a small bug?
I would be really grateful for any help.

Cheers,
Anton.

**Comment by bolatovumar** [comment, marked as answer]
Hi @hidaloop,

This seems to actually be a bug. We show these lightning things independent on whether a particular network supports LN, these things are basically hardcoded. I made a PR here to address this issue: https://github.com/btcpayserver/btcpayserver/pull/3935

**Reply by hidaloop** [reply]
Merci beaucoup for getting back to me. I will fetch upstream and give it a look.
But I am still marking it as answered.

Regards, Anton.
